redbot/cogs/mutes/mutes.py

```diff
@@ -0,0 +1,1378 @@
+
+        Examples:
+        `[p]mute @member1 @member2 spam 5 hours`
+        `[p]mute @member1 3 days`
+
+        """
+        if not users:
+            return await ctx.send_help()
+        if ctx.me in users:
+            return await ctx.send(_("You cannot mute me."))
+        if ctx.author in users:
+            return await ctx.send(_("You cannot mute yourself."))
+        if not await self._check_for_mute_role(ctx):
+            return
+        async with ctx.typing():
+            duration = time_and_reason.get("duration", {})
+            reason = time_and_reason.get("reason", None)
+            time = ""
+            until = None
+            if duration:
+                until = datetime.now(timezone.utc) + timedelta(**duration)
+                time = _(" for {duration}").format(
+                    duration=humanize_timedelta(timedelta=timedelta(**duration))
+                )
+            else:
+                default_duration = await self.config.guild(ctx.guild).default_time()
+                if default_duration:
+                    until = datetime.now(timezone.utc) + timedelta(**default_duration)
+                    time = _(" for {duration}").format(
+                        duration=humanize_timedelta(timedelta=timedelta(**default_duration))
+                    )
+            author = ctx.message.author
+            guild = ctx.guild
+            audit_reason = get_audit_reason(author, reason)
+            success_list = []
+            issue_list = []
+            for user in users:
+                success, issue = await self.mute_user(guild, author, user, until, audit_reason)
+                if issue:
+                    issue_list.append((user, issue))
+                if success:
+                    success_list.append(user)
+                    await modlog.create_case(
+                        self.bot,
+                        guild,
+                        ctx.message.created_at.replace(tzinfo=timezone.utc),
+                        "smute",
+                        user,
+                        author,
+                        reason,
+                        until=until,
+                        channel=None,
+                    )
+            if success_list:
+                if ctx.guild.id not in self._server_mutes:
+                    self._server_mutes[ctx.guild.id] = {}
+                msg = _("{users} has been muted in this server{time}.")
+                if len(success_list) > 1:
+                    msg = _("{users} have been muted in this server{time}.")
+                await ctx.send(
+                    msg.format(users=humanize_list([f"{u}" for u in success_list]), time=time)
+                )
+            if issue_list:
+                message = _(
+                    "{users} could not be muted in some channels. "
+                    "Would you like to see which channels and why?"
+                ).format(users=humanize_list([f"{u}" for u, x in issue_list]))
```

Comment:
Message written like that only makes sense for overrides-based mutes, not so much for role-based mutes.